aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -0,0 +1,139 @@
+
+![Kestrel to Internet](kestrel/_static/kestrel-to-internet.png)
+
+The most important reason for using a reverse proxy for edge deployments (exposed to traffic from the Internet) is security. Kestrel is relatively new and does not yet have a full complement of defenses against attacks. This includes but isn't limited to appropriate timeouts, size limits, and concurrent connection limits.
+
+Another scenario that requires a reverse proxy is when you have multiple applications that share the same port running on a single server. That doesn't work with Kestrel directly because Kestrel doesn't support host-header routing. When you configure Kestrel to listen on a port, it handles all traffic for that port regardless of host header.
+
+Even if a reverse proxy server isn't required, you can use one to improve performance. A reverse proxy can let you offload some work from your application server, such as serving static content, and it can cache and compress responses. A reverse proxy can also simplify load balancing and SSL set-up -- only your reverse proxy server requires an SSL certificate, and that server can communicate with your application servers on the internal network using plain HTTP.
+
+## How to use Kestrel in ASP.NET Core apps
+
+Install the [Microsoft.AspNetCore.Server.Kestrel](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.Kestrel/) NuGet package.
+
+Call the [`UseKestrel`](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilderKestrelExtensions/index.html#Microsoft.AspNetCore.Hosting.WebHostBuilderKestrelExtensions.UseKestrel.md) extension method on `WebHostBuilder` in your `Main` method, specifying any [Kestrel options](https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Server/Kestrel/KestrelServerOptions/) that you need, as shown in the following example:
+
+[!code-csharp[](kestrel/sample/Program.cs?name=snippet_Main&highlight=13-19)]
+
+### URL prefixes
+
+By default Kestrel binds to `http://localhost:5000`. You can configure URL prefixes and ports for Kestrel to listen on by using the `UseURLs` extension method, the `server.urls` command-line argument, or the ASP.NET Core configuration system. For more information about these methods, see [Hosting](../../fundamentals/hosting.md). For information about how URL binding works when you use IIS as a reverse proxy, see [ASP.NET Core Module](aspnet-core-module.md). 
+
+URL prefixes for Kestrel can be in any of the following formats. 
+
+* IP address with port number
+
+  ```
+  http://65.55.39.10:80/
+  https://65.55.39.10:443/
+  ```
+
+  Address 0.0.0.0 is a special case that binds to all IP addresses.
+
+
+* IPV6 address with port number
+
+  ```
+  http://[0:0:0:0:0:ffff:4137:270a]:80/ 
+  https://[0:0:0:0:0:ffff:4137:270a]:443/ 
+  ```
+
+* Host name with port number
```

Comment:
Host names, *, and +, are not special, anything that is not a recognized IP address or localhost will bind to all IPv4 and IPv6 IPs.